tests/test_transform.py

```diff
@@ -0,0 +1,313 @@
+
+def test_lighting():
+    im = np.array([
+        [0.,   0.1,  0.,  ],
+        [0.01, 0.2,  0.03,],
+        [0.,   0.3,  0.,  ],])
+    e = im
+    a = lighting(im, 0, 1)
+    # TODO: better test taht allows for visual inspection
+    np.testing.assert_array_equal(a, e)
+    e =np.array([[0.5 , 0.6 , 0.5 ],
+                [0.51, 0.7 , 0.53],
+                [0.5 , 0.8 , 0.5 ]], dtype=np.float32)
+    a = lighting(im, 0.5, 1)
+    np.testing.assert_array_equal(a, e)
+@pytest.mark.skip(reason="It does not work for some reason see #431")
+def test_rotate_cv():
+    im = np.array([
+        [0.,   0.1,  0., ],
+        [0.,   0.2,  0., ],
+        [0.,   0.3,  0., ],])
+    a = rotate_cv(im, 90)
+    # TODO: find out why this test breaks
+    e = np.array([[0. , 0. , 0. ],
+                  [0.1, 0.2, 0.3],
+                  [0. , 0. , 0. ],])
+    np.testing.assert_array_equal(a, e)
+
+@pytest.mark.skip(reason="It does not work for some reason see #431")
+def test_rotate_cv_vs_dihedral():
+    im = np.array([
+        [0.,   0.1,  0., ],
+        [0.,   0.2,  0., ],
+        [0.,   0.3,  0., ],])
+    a = rotate_cv(im, 180)
+    e = dihedral(im, 6)
+    np.testing.assert_array_equal(a, e)
+    
+def test_no_crop():
+    im = np.array([
+        [0.,   0.1,  0., ],
+        [0.,   0.2,  0., ],])
+    a = no_crop(im, 4)
+    e = np.array([[0. , 0.066 , 0.066, 0 ],
+                  [0,   0.066,  0.066, 0 ],
+                  [0. , 0.133 , 0.133, 0 ],
+                  [0. , 0.133 , 0.133, 0 ]])
+    np.testing.assert_array_almost_equal(a, e, decimal=3)
+
+def test_center_crop():
+    im = np.array([
+        [0.,   0.1,  0.,  ],
+        [0.01, 0.2,  0.03,],
+        [0.,   0.3,  0.,  ],])
+    a = center_crop(im, 1)
+    e = np.array([[0.2]])
+    np.testing.assert_array_equal(a, e)
+    im = np.array([
+        [0.,   0.1,  0.,  0],
+        [0.01, 0.2,  0.9, 0.04],
+        [0.,   0.3,  0.,  0],])
+    a = center_crop(im, 1)
+    e = np.array([[0.9]])
+    np.testing.assert_array_equal(a, e)
+
+def test_googlenet_resize():
+    # TODO: figure out how to test this in a way it make sense
+    pass
+
+def test_cutout():
+    im = np.ones([128,128,3], np.float32)
+    with_holes = cutout(im, 1, 10)
+    assert (with_holes == 0).sum() == 300, "There is one cut out hole 10px x 10px in size (over 3 channels)"
+
+def test_scale_to():
+    h=10
+    w=20
+    ratio = 127./h
+    assert scale_to(h, ratio, 127) == 127
+    assert scale_to(w, ratio, 127) == 254
+
+def test_crop():
+    im = np.ones([128,128,3], np.float32)
+    assert crop(im, 1, 1, 10).shape == (10,10,3)
+
+def test_to_bb():
+    im = np.array([[0, 0, 0, 0, 0],
+                   [0, 1, 1, 1, 0],
+                   [0, 1, 1, 1, 0],
+                   [0, 1, 1, 1, 0],
+                   [0, 0, 0, 0, 0], ], dtype=np.float32)
+    expect = [1,1,3,3]
+    np.testing.assert_array_equal(to_bb(im, "not used"), expect)
+
+### tests for transformation objects
+def test_RandomCrop():
```

Comment:
(1) Sorry for leaving my last comments on the commit and not here...
(2) One last thought is that this violates method naming conventions in general, not sure if we care about that vs the class name.

Reply:
Sticking to PEP8 around naming test functions is bad idea as the functions names are used in different context.
The most important thing is that you know what the tests are about when you read the function names.
I like bdd specification style of test (wiki)[https://en.wikipedia.org/wiki/Behavior-driven_development#Specialized_tooling_support], (ScalaTest as example)[http://www.scalatest.org], where you write a spec in plain English, if you know any good way of doing such thing in pytest please share.

In py.test I tend to follow this convention:

test_<object_of_test>__<what function is being tested> 

But I'm not sure consistent around that (I've forgotten to put double underscore to separate object from a feature that is being tested), I'm fine to adopt some good convention.

Reply:
We're trying to follow the convention from the Python testing guide:

>Use long and descriptive names for testing functions. The style guide here is slightly different than that of running code, where short names are often preferred. The reason is testing functions are never called explicitly. square() or even sqr() is ok in running code, but in testing code you would have names such as test_square_of_number_2(), test_square_negative_number(). These function names are displayed when a test fails, and should be as descriptive as possible.

See also: https://github.com/fastai/fastai/blob/master/docs/testing.adoc